src/experiment/api.ts

```diff
@@ -24,11 +23,90 @@ export async function getExperimentExecution(path: string): Promise<ExecutionRes
     const response = await executeApiCall({
       method: 'GET',
       path,
-      fullyQualifiedUrl: true,
+      fullyQualifiedUrl: true
     });
     return (await response.json()) as ExecutionResult;
   } catch (e) {
-    const error = await abortExecutionWithError(e, 'Failed to get experiment run ');
-    throw error;
+    throw await abortExecutionWithError(e, 'Failed to get experiment run ');
+  }
+}
+
+
+export async function fetchExperiment(key: string): Promise<Experiment> {
+  try {
+    const response = await executeApiCall({
+      method: 'GET',
+      path: `/api/experiments/${key}`
+    });
+    return response.json();
+  } catch (e: any) {
+    if (e.response?.status === 404) {
+      throw await abortExecution('Experiment %s not found.', key);
```

Comment:
**Issue:** `abortExecution` does not return a `Promise` and hence doesn't need to be awaited on. Curious that TypeScript doesn't complain about it 🤔.

Although there should be a warning:

<img width="829" alt="image" src="https://user-images.githubusercontent.com/596443/198199554-6730ebda-319f-453f-a2ad-b39e2a06f1d8.png">